grakn-graql/src/main/java/ai/grakn/graql/internal/pattern/property/IdProperty.java

```diff
@@ -71,24 +70,16 @@ public String getProperty() {
     }
 
     @Override
-    public void insert(Var var, InsertQueryExecutor executor) throws GraqlQueryException {
-        executor.builder(var).id(id());
+    public PropertyExecutor insert(Var var) throws GraqlQueryException {
```

Comment:
You know after seeing this over and over again I am wondering if this abstraction, while handy, will come to bite us during debugging. This maybe insanely verbose but maybe it will be better for `PropertyExecutor.builder(....)` to accept an specific interface rather than a generic consumer? That way we can have explicit implementations of that interface which might be easier to debug.